lambeq/backend/drawing/drawing.py

```diff
@@ -144,17 +148,56 @@ def draw(diagram: Diagram, **params) -> None:
             backend = _draw_controlled_gate(backend, drawable, node, **params)
         elif not drawn_as_spider(node.obj):
             backend = _draw_box(backend, drawable, node, **params)
-
+        else:
+            wire_drawings = compute_spider_wires(node, drawable, **params)
+            backend.draw_spider(node,wire_drawings,**params)
```

Comment:
@Ragunath1729 aren't you changing the `draw_spider` signature here too much? I expect the arguments for a function called `draw_spider` to be a `BoxNode`, at least.

What you can do is something like (pseudocode):

```python
for each spider:
    spider_colors = get_spider_leg_colors(node)
    backend.draw_spider(node, spider_colors)
```

Then, the `draw_spider` function will do what's currently happening inside your `compute_spider_wires` function.

Reply:
Or better yet, implement the `_get_..._color` functions inside the `DrawingBackend` abstract class as we've discussed